java/vector/src/main/java/org/apache/arrow/vector/dictionary/Dictionary.java

```diff
@@ -52,14 +52,6 @@ public String toString() {
   }
 
   @Override
-  public boolean equals(Object o) {
```

Comment:
hmm, so basically `FieldVector.equals` isn't correct? I'm not sure if it would affect anything to remove this or not...